src/test/java/com/desiato/puresynth/controllers/UserControllerTest.java

```diff
@@ -0,0 +1,153 @@
+
+    @Test
+    public void createUser_ShouldReturnCreatedUser() throws Exception {
+        String uniqueEmail = generateUniqueEmail();
+        String newUserJson = String.format("""
+        {
+            "email": "%s",
+            "password": "password123"
+        }
+        """, uniqueEmail);
+
+        mockMvc.perform(post("/api/user")
+                        .contentType(MediaType.APPLICATION_JSON)
+                        .content(newUserJson))
+                .andExpect(status().isCreated())
+                .andExpect(jsonPath("$.email").value(uniqueEmail))
+                .andDo(print());
+    }
+
+    @Test
+    public void deleteUser_ShouldDeleteUserAndReturnOk() throws Exception {
+        // Given
+        String email = generateUniqueEmail();
+        String password = "password123";
+        User existingUser = userService.createUser(email, passwordEncoder.encode(password));
+        Long userId = existingUser.getId();
+
+        Optional<String> optionalToken = authService.authenticate(email, password);
+        assertTrue(optionalToken.isPresent(), "Authentication failed, no token obtained.");
+        String token = optionalToken.get();
+
+        userService.deleteUserSessions(userId);
+
+        // When
+        mockMvc.perform(delete("/api/user/" + userId)
+                        .header("Auth-Token", token))
+                .andExpect(status().isOk());
+
+        // Then
+        // Check that the user is deleted
+        mockMvc.perform(get("/api/user/" + userId))
+                .andExpect(status().isNotFound())
+                .andReturn();
+    }
+
+    @Test
+    public void getUser_ShouldReturnUser() throws Exception {
+        User existingUser = userService.createUser(generateUniqueEmail(),
+                passwordEncoder.encode("password123"));
+
+        Optional<String> optionalToken = authService.authenticate(existingUser.getEmail(), "password123");
+        assertTrue(optionalToken.isPresent());
+        String token = optionalToken.get();
+
+        mockMvc.perform(get("/api/user/" + existingUser.getId())
+                        .header("Auth-Token", token))  // Add token to request header
+                .andExpect(status().isOk())
+                .andExpect(jsonPath("$.id").value(existingUser.getId()))
+                .andExpect(jsonPath("$.email").value(existingUser.getEmail()));
+    }
+
+    @Test
+    public void updateUser_ShouldUpdateUserAndReturnUpdatedUser() throws Exception {
+        // Given
+        User existingUser = userService.createUser(generateUniqueEmail(),
+                passwordEncoder.encode("password123"));
+        Long userId = existingUser.getId();
+
+        // Authenticate to obtain token
+        Optional<String> optionalToken = authService.authenticate(existingUser.getEmail(), "password123");
+        assertTrue(optionalToken.isPresent());
+        String token = optionalToken.get();
```

Comment:
Lines 94-101 should be moved to a utility function. You're calling it in each test, where you need to create an authenticated user. Alternatively, you can look how to create some sort of test-only interceptor to always accept tokens for tests, so that you don't need to authenticate each test requests (apart from purely auth tests).